### PR TITLE
fix #4907 feat(nimbus): reject experiment api

### LINKED ISSUE #4907: Create reject experiment mutation
To allow the UI to submit a rejection for an experiment we need to expose a rejection mutation in the V5 API.

This mutation should take in an experiment id and a rejection message and then update the experiment publish_status from review back to idle with a new changelog that includes the rejection message in its message field.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1116)
┆Issue Number: EXP-1116

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* The frontend needs to be able to reject a launching or ending experiment

This commit

* Adds a changelog_message field to the V5 API that gets attached to the changelog which can be used to store a rejection reason
* Rejecting from the frontend will involve a mutation that sets publish_status back to IDLE, sets is_end_requested to False, and attaches a changelog_message
* Updates the V5 serializer to allow status, publish_status, and is_end_requested to be mutated at any status
* Because we have to mutate is_end_requested, this supports the end request use case as well so we can fold the EndMutation logic back into the UpdateExperiment mutation
* Updates frontend to call UpdateExperiment instead of EndExperiment with isEndRequested

### CODE REVIEW

**Comment by jaredlockhart** [comment]
Blocked on #4850

**Comment by lmorchard** [comment]
> Rejecting from the frontend will involve a mutation that sets publish_status back to IDLE, sets is_end_requested to False, and attaches a changelog_message

Hmm, I know this PR drops the EndExperiment mutation. But, this suggests to me maybe we should *add* a RejectReviewExperiment mutation to contain complexity on the server-side.

Granted, the complexity is just changing three fields and the generic `UpdateMutation` is attractively simple. But just thinking about how distributed across the client/server boundary this sort of logic / data management should get - or if the back-end should have most of the smarts in one place.

**Review by lmorchard — APPROVED** [review]
LGTM with maybe one philosophical question and a code-golfing comment

File: app/experimenter/experiments/api/v5/serializers.py [hunk 1]
```diff
@@ -213,14 +213,17 @@ def validate(self, data):
             for status_field, restricted_statuses in restrictive_statuses.items():
                 current_status = getattr(self.instance, status_field)
                 is_locked = current_status not in restricted_statuses
-                is_modifying_other_fields = set(data.keys()) != {status_field}
-                if is_locked and is_modifying_other_fields:
+                modifying_fields = set(data.keys()) - set(
+                    NimbusExperiment.STATUS_UPDATE_EXEMPT_FIELDS
+                )
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply app/experimenter/experiments/api/v5/serializers.py:218]: Yeah that's an intersting idea.  We'll probably have to do that once users can start editing live experiments, but only certain fields.

**Comment by jaredlockhart** [comment]
> > Rejecting from the frontend will involve a mutation that sets publish_status back to IDLE, sets is_end_requested to False, and attaches a changelog_message
> 
> Hmm, I know this PR drops the EndExperiment mutation. But, this suggests to me maybe we should _add_ a RejectReviewExperiment mutation to contain complexity on the server-side.
> 
> Granted, the complexity is just changing three fields and the generic `UpdateMutation` is attractively simple. But just thinking about how distributed across the client/server boundary this sort of logic / data management should get - or if the back-end should have most of the smarts in one place.

Yeah also a super valid point.  There might be some kind of middle ground where we can define them as constants somehow.... like

```py
const LaunchReviewState = {
    status: NimbusExperimentStatus.Draft,
    publishStatus: NimbusExperimentPublishStatus.Review,
};

const endRequestState = {
    status: NimbusExperiment.Status.Live,
    publishStatus: NimbusExperiment.PublishStatus.Review,
    isEndRequested: true,
};

updateExperiment(LaunchReviewState);
updateExperiment(EndRequestState);
```

etc

The good news is with every field/value being strongly enforced all the way from server to client we can at least make sure we keep these things in sync.

**Comment by jodyheavener** [comment]
@jaredlockhart I've rebased/resolved conflicts on this, and am just going through the flow now. Anything else that this PR notably needed updating on?

**Review by jodyheavener — APPROVED** [review]
Heck ya

**Pull request merged**